### DISCUSSION TITLE
Dynamic runtime migrations

### DISCUSSION THREAD

**Discussion by Zyles** [mikro-orm/mikro-orm#2326] [Q&A]
**Is your feature request related to a problem? Please describe.**

Can MikroORM create, alter and drop tables during runtime without code?

I am building a very generic system where a user can create an entity, add columns/data types and it should then create this table for them where they can CRUD data. My routes in NestJS are dynamic and the CRUD endpoints are created based on the entity.

I don't really want to use a EAV system for this but rather create unique tables every time the user creates an entity. I would prefer being able to look directly into the database row values instead of doing massive joins.

Is this possible to implement? Or do I need to generate the entities files, write them to disk and migrate them with a bit of a hack?

**Describe the solution you'd like**

A way to create/alter/delete tables during runtime without TS classes / or generate TS classes from JSON?

**Describe alternatives you've considered**

EAV tables.

**Comment by B4nan** [comment]
How would you want to work with the tables? Currently you need to have an entity, entity is a class. You can register new entities dynamically (in v5 there is a helper method on `MikroORM` class, in v4 it can work too but with a bit more code), as well as you could use `EntitySchema` to create new entity metadata instance on runtime, without having a class. You can also use `SchemaGenerator` during runtime (I don't think migrations are a fit for this use case, they are basically just a layer on top of `SchemaGenerator`). 

So technically it should be possible, but I am not sure how would you expect to work with the entities, where would you store the information about such entities that don't have a regular class on the file system (e.g. what would happen on app restart when you loose the in memory information).
